File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(x, computers):
    idx = 0
    for i in computers[x]:
        if idx != x and i == 1:
            computers[x][idx] = 0
            computers[idx][x] = 0
            dfs(idx, computers)
        idx += 1
    return

def solution(n, computers):
    answer = 0
    for i in computers:
        if sum(i) == 1:
            answer += 1
    for i in range(n):
        for j in range(n):
            if i != j and computers[i][j] == 1:
                computers[i][j] = 0
                computers[j][i] = 0
                dfs(j, computers)
                answer += 1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    static boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                visited[i] = true;
                bfs(n, computers, i);
                answer++;
            }
        }
        return answer;
    }
    
    public static void bfs(int n, int[][] computers, int i) {
        for (int idx = 0; idx < n; idx++) {
            if (!visited[idx] && computers[idx][i] == 1) {
                visited[idx] = true;
                bfs(n, computers, idx);
            }
        }
        return ;
    }
}